homeassistant/components/google_drive/application_credentials.py

```diff
@@ -15,9 +18,16 @@ async def async_get_authorization_server(hass: HomeAssistant) -> AuthorizationSe
 
 async def async_get_description_placeholders(hass: HomeAssistant) -> dict[str, str]:
     """Return description placeholders for the credentials dialog."""
+    redirect_url = (
+        MY_AUTH_CALLBACK_PATH
+        if "my" in hass.config.components
+        else f"{hass.config.external_url}{AUTH_CALLBACK_PATH}"
+        if hass.config.external_url
+        else "N/A"
```

Comment:
If it's "N/A" the instructions in the translation will be incorrect.

Reply:
I think the beauty of the way the oauth helper calculates the URL is that we know the user is able to access the link at that place at that time. I'm wondering if there's some way to still get that link

Reply:
I changed it to `https://YOUR_DOMAIN:PORT/auth/external/callback` so that users can still see what format they should put their redirect URL and I added tests. @joostlek I haven't been able to find a way.  This really only matters for the very few advanced users that have the 'my' integration disabled so I don't want to invest more time trying to figure it out.